server/report_builder/src/build.rs

```diff
@@ -236,6 +240,36 @@ fn make_report(args: &BuildArgs, mut files: HashMap<String, PathBuf>) -> Result<
         entries.insert(name, value);
     }
 
+    // first check if there is nominated path to custom wasm function
+    if let Some(custom_wasm_function) = &args.custom_wasm_function {
+        let encoded = BASE64_STANDARD.encode(fs::read(custom_wasm_function).unwrap());
+        index.convert_data = Some(encoded)
+    };
+
+    // Then look for data conversion with js functions
+    if let Some(convert_data) = &args.convert_data {
+        if index.convert_data.is_none() {
+            let js = &format!("./{}/convert_data.js", convert_data);
+
+            let ts = &format!("./{}/convert_data.d.ts", convert_data);
+
+            let wasm = &format!("./{}/convert_data.wasm", convert_data);
+
+            let dir_path = &format!("./{}/", convert_data);
+
+            Command::new(&format!("ls {}", dir_path));
+
+            Command::new("extism-js")
+                .args([&js, "-i", &ts, "-o", &wasm])
+                .output()
+                .unwrap();
+
+            let encoded = BASE64_STANDARD.encode(fs::read(wasm).unwrap());
+
+            index.convert_data = Some(encoded)
+        }
+    }
+
```

Comment:
I'm wondering if here it would be better to add consumer focussed generation of paths so users only need to add the file name for example rather than adding the whole file path in the manifest.json